blog/12-25-22-best-practices-for-writing-smart-contracts-with-real-world-examples/best-practices-for-writing-smart-contracts-with-real-world-examples.md

```diff
@@ -0,0 +1,534 @@
+
+-	contracts
+	-	match-making
+		- interfaces
+			- Common.sol
+			- IHusbandToBe.sol
+			- IWifeTobe.sol
+			- IParent.sol
+		-	wealth
+			-	CashInBank.sol
+			-	Properties.sol
+		- HusbandToBe.sol
+		- WifeToBe.sol
+		- Parent.sol
+
+Our contract code size is a bit bigger and we have successfully broken them down to implement a modular style.
+
+## Circuit breaker
+
+This acts as a stopper or a switch that breaks executions. In the event a bug is detected in your contract or a potential vulnerability has adversely affected the dApp, you need a way to mitigate or prevent a huge loss from further happening. Halting contract execution may be very effective. Usually, this would be a function that when invoked, would simultaneously prevent execution of other functions. The openzeppelin “Pausable” module is an example of a circuit breaker. When your contract inherits from it, you will need to activate it by overriding the “_pause()” and “_unpause()” functions so that only authorized accounts e.g the "owner" can call it. In our case, we will need each of the subjects to be able to have some level of control over their respective profiles soon as they subscribed in order to prevent reserved functions from being invoked by anyone. Ddoing so, we will have a reasonable approval mechanism among the parties that regard and respect each party’s opinion such as the WifeToBe can express her personal feelings even while the parents feel certain things are right for her. In Africa, we’ve had countless cases where parents force their daughters to marry men they do not wish to marry. Having such a tamper-proof system as the Celo blockchain to balance, checkmate preferences and respect opinions would go a long way to make the world a better place. We are going to leverage the openzeppelin’s “security” and “access” modules to implement a circuit breaker that pauses the contract when invoked by the authorized account i.e. "owner". Copy/paste the following code at the top of  each standalone deployable contract file.
```

Comment:
```suggestion
This acts as a stopper or a switch that breaks executions. In the event a bug is detected in your contract or a potential vulnerability has adversely affected the dApp, you need a way to mitigate or prevent a huge loss from further happening. Halting contract execution may be very effective. Usually, this would be a function that when invoked, would simultaneously prevent execution of other functions. The openzeppelin “Pausable” module is an example of a circuit breaker. When your contract inherits from it, you will need to activate it by overriding the “_pause()” and “_unpause()” functions so that only authorized accounts e.g the "owner" can call it. In our case, we will need each of the subjects to be able to have some level of control over their respective profiles soon as they subscribed in order to prevent reserved functions from being invoked by anyone. Doing so, we will have a reasonable approval mechanism among the parties that regard and respect each party’s opinion such as the WifeToBe can express her personal feelings even while the parents feel certain things are right for her. In Africa, we’ve had countless cases where parents force their daughters to marry men they do not wish to marry. Having such a tamper-proof system as the Celo blockchain to balance, checkmate preferences and respect opinions would go a long way to make the world a better place. We are going to leverage the openzeppelin’s “security” and “access” modules to implement a circuit breaker that pauses the contract when invoked by the authorized account i.e. "owner". Copy/paste the following code at the top of  each standalone deployable contract file.
```